bioblend/_tests/TestGalaxyJobs.py

```diff
@@ -33,6 +44,57 @@ def test_run_and_rerun_random_lines(self):
         self.assertEqual(build_for_rerun['state_inputs']['seed_source']['seed'], 'asdf')
 
         rerun_output = self.gi.jobs.rerun_job(original_job_id)
-        rerun_output_content = self.gi.datasets.download_dataset(rerun_output['outputs'][0]['id'])
+        rerun_job_id = rerun_output['jobs'][0]['id']
+        self._wait_job(original_job_id)
         original_output_content = self.gi.datasets.download_dataset(original_output['outputs'][0]['id'])
+        self._wait_job(rerun_job_id)
+        rerun_output_content = self.gi.datasets.download_dataset(rerun_output['outputs'][0]['id'])
         self.assertEqual(rerun_output_content, original_output_content)
+
+    @test_util.skip_unless_galaxy('release_21.01')
+    @test_util.skip_unless_tool("Show beginning1")
+    def test_rerun_and_remap(self):
+        path = test_util.get_abspath(os.path.join('data', 'select_first.ga'))
+        wf = self.gi.workflows.import_workflow_from_local_path(path)
+        wf_inputs = {
+            "0": {'src': 'hda', 'id': self.dataset_id},
+            "1": "-1",
+        }
+        inv = self.gi.workflows.invoke_workflow(wf['id'], wf_inputs)
+        for _ in range(120):
+            inv = self.gi.invocations.show_invocation(inv['id'])
+            if inv['steps']:
+                for step in inv['steps']:
+                    if step['job_id']:
+                        self._wait_job(step['job_id'])
+                break
+            time.sleep(.5)
+        new_history_id = inv['history_id']
+        history_contents = self.gi.histories.show_history(new_history_id, contents=True)
+        self.assertEqual(len(history_contents), 2)
+        self.assertEqual(history_contents[0]['state'], 'error')
+        self.assertEqual(history_contents[1]['state'], 'paused')
+
+        # now rerun and remap with correct input param
+        job_id = self.gi.datasets.show_dataset(history_contents[0]['id'])['creating_job']
+        tool_inputs_update = {
+            'lineNum': '1'
+        }
+        self.gi.jobs.rerun_job(job_id, remap=True, tool_inputs_update=tool_inputs_update)
+        jobs_to_complete = [job['id'] for job in self.gi.jobs.get_jobs() if job['history_id'] == new_history_id]
+        for job_id in jobs_to_complete:
+            self._wait_job(job_id)
+
+        # check if workflow output has completed
+        workflow_output = [dataset for dataset in self.gi.histories.show_history(new_history_id, contents=True) if dataset['name'] == 'paste_output'][0]
+        self.assertEqual(workflow_output['state'], 'ok')
+        self.assertEqual(workflow_output['hid'], 2)
+        self.assertEqual(workflow_output['id'], history_contents[1]['id'])
+        self._wait_and_verify_dataset(workflow_output['id'], b'line 1\tline 1\n')
+
+    def _wait_job(self, job_id):
```

Comment:
Move to `bioblend/galaxy/jobs/__init__.py` and expose it?

Reply:
I've added a longer version of this method to `JobsClient` in [`367bbef` (#369)](https://github.com/galaxyproject/bioblend/pull/369/commits/367bbefac0d7a8eb2621cc160bed166275720a75)